pkg/cluster/operation/operation.go

```diff
@@ -27,6 +28,24 @@ const (
 	EnvNamePDEndpointOverwrite  = "FORCE_PD_ENDPOINTS"
 )
 
+type contextKey string
+
+// ContextKeyCurrentVersion is the key of current cluster version
+const ContextKeyCurrentVersion contextKey = "currentVersion"
+
+func getCurrentVersionFromContext(ctx context.Context) string {
```

Comment:
I suggest just put cluster version on args of upgrade function. There is no need to put it to context. rest LGTM

Reply:
I've tried it, but it causes the signature to change, which makes a lot of other code changes.
 If it's acceptable, it's ok to me.

Reply:
I think it is acceptable, since we never promise api stable in tiup v1.x

Reply:
resolved